feat: tabs for scenes

rostyslavvnahornyi opened the pull request.

kfarr committed:
hotfix support new json utils deployed on 3dstreet

kfarr committed:
npm run dist and firebase deploy

kfarr committed:
whoops at least it was just dev

rostyslavvnahornyi committed:
feat: tabs for scenes

kfarr committed:
Merge branch 'master' into issue#249

kfarr commented:
thanks @rostyslavvnahornyi I did [some updates](https://github.com/3DStreet/3dstreet-editor/pull/278/commits/3f8225592bd2b4187299a0221d994be850af0285) and the last thing I need your help with is 
- [x] - the tab component should be fixed as a header, not scroll along with the scenes


(the below is already done, just keeping here FYI:)
- there appears to be an issue with the existing subscription to 'subscribeToUserScenes' -- if the user selects community tab, and there is a change to a user scene, then the results are overwritten [1]
- each switch of a tab results in a new query without pagination; this means firebase is essentially returning the entire database including the data field in each response which means it is a huge download each time and takes a few seconds and we do not have a loading state, so this is confusing to users.

suggestion: (done in commit https://github.com/3DStreet/3dstreet-editor/pull/278/commits/3f8225592bd2b4187299a0221d994be850af0285)
- instead, what if we create 2 scene card components in separate divs that load at modal open, each subscribed to user scenes and community scenes respectively; the tab only switches which div is visible, it does not affect any query since both divs populate upon load. This way loading only happens once for each, and both use subscriptions to reflect changes.

[1] video
https://github.com/3DStreet/3dstreet-editor/assets/470477/9fd4237f-22aa-4247-88f8-63d51d9801f2

kfarr committed:
refactor fetching community and owned scenes

- don't use subscribe, instead refetch when modal is re-opened
- use 2 queries displayed on separate tabs made visible by tab chooser
- remove unused subscribeToUserScenes code

kfarr committed:
move tab to header and fix styles

kfarr committed:
simplify card switching logic
